### PR TITLE
Load Redis test endpoints from config file or env vars

### PR DESCRIPTION
**Pull request by uglide**

### CODE REVIEW

File: tests/NRedisStack.Tests/RedisFixture.cs [hunk 1]
```diff
@@ -18,7 +59,42 @@ public RedisFixture()
...
-        Redis = Connect(clusterConfig, out isOSSCluster);
+
+        if (redisEndpointsPath != null && File.Exists(redisEndpointsPath))
+        {
+            string json = File.ReadAllText(redisEndpointsPath);
+            var parsedEndpoints = JsonSerializer.Deserialize<Dictionary<string, EndpointConfig>>(json);
+
+            redisEndpoints = parsedEndpoints ?? throw new Exception("Failed to parse the Redis endpoints configuration.");
+        }
+        else
```
Comment (gerzse) [inline tests/NRedisStack.Tests/RedisFixture.cs:70]: Just an idea, would it work to have a default JSON file to load, so we don't do any config from code?

Comment (uglide) [reply tests/NRedisStack.Tests/RedisFixture.cs:70]: Yes, we can replace it with a default JSON file. I did it only to keep it backward compatible with the existing GitHub workflows.

File: tests/NRedisStack.Tests/RedisFixture.cs [hunk 2]
```diff
@@ -1,13 +1,54 @@
...
+        {
+            configurationOptions.EndPoints.Add(endpoint);
+        }
+
+        if (password != null)
+        {
+            configurationOptions.Password = password;
+        }
+
+        // TODO(imalinovskiy): Add support for TLS
```
Comment (tishun) [inline tests/NRedisStack.Tests/RedisFixture.cs:32]: Can we perhaps list somewhere the environments we would like to have?

I think the existing tests (which make sense to us) are a good reference as to what we need.

Everybody can suggest additions / deletions to this list

Comment (uglide) [reply tests/NRedisStack.Tests/RedisFixture.cs:32]: The list of required environments depends on the current test coverage. For example, in Jedis, we need a huge list of Redis endpoints https://github.com/redis/jedis/pull/3836/files#diff-c731b96d111ab92ab2f88dcbe2e36abef53c269a71f96598463e6cc3b70b7acd and it doesn't yet cover the Sentinel and Cluster endpoints. As a first stage, I suggest identifying those endpoints and collecting them in endpoint.json file in each repo.

Comment (tishun) [reply tests/NRedisStack.Tests/RedisFixture.cs:32]: Yes, this is what I meant, gather a list so we know what we need. Thanks!

**Review by atakavci — APPROVED** [review]
lgtm!

**Pull request merged**